connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/util/RemoteSparkSession.scala

```diff
@@ -114,19 +118,25 @@ object SparkConnectServerUtils {
 }
 
 trait RemoteSparkSession extends ConnectFunSuite with BeforeAndAfterAll {
-  import SparkConnectServerUtils._
   var spark: SparkSession = _
+  var server: SparkConnectServer = _
+
+  protected def sparkConf: SparkConf = new SparkConf(loadDefaults = false)
 
   override def beforeAll(): Unit = {
     super.beforeAll()
-    SparkConnectServerUtils.start()
-    spark = SparkSession.builder().client(SparkConnectClient.builder().port(port).build()).build()
+    server = new SparkConnectServer(sparkConf)
```

Comment:
This would make each test suite who extend the RemoteSparkSession start a new server. The server start time is pretty bad ~1 min.

Can we have another impl of RemoteSparkSession e.g. `RemoteSparkSessionWithConf` which take extra configs and start new server on demands. But leave the original `RemoteSparkSession` share one server only?

Reply:
+1 to not impact existing test cases that do not need the SparkConf.